sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import scala.collection.immutable.Seq
+
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.sql.types.{MapType, StringType}
+
+// scalastyle:off nonascii
+class CollationSQLExpressionsSuite
```

Comment:
We already have multiple test suite for string collation, can you justify the need of a new suite?

Reply:
We do have separate suites for `stringExpressions` and `regexpExpressions`, which are the two larger "groups" of SQL expressions. So now it's a bit unclear on where to put these tests, and there may be more to come for some expression that don't fall into any of the previous 2 categories

So it's either multiple smaller suites or 1 large suite to cover them all. Should we go with this for now and then I'll shuffle these tests around in a separate effort in order to unify them under a single suite?

Reply:
for example, Mihailo is working on some more expressions right now, and they don't fall under stringExpressions nor regexpExpressions, so I suggest we keep `CollationSQLExpressionsSuite.scala` for these purposes

and then we can migrate CollationStringExpressionsSuite & CollationRegexpExpressionsSuite into CollationSQLExpressionsSuite later on